client/state/happychat/selectors.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import isArray from 'lodash/isArray';
+import map from 'lodash/map';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+
+/**
+ * Gets the current happychat connection status
+ * @param {Object} state - global redux state
+ * @return {String} current state value
+ */
+export const getHappychatConnectionStatus = createSelector(
+	state => state.happychat.status,
+	state => state.happychat.status
+);
+
+// In previous implementations timeline events were stored as an array of [ 'message text', { ... properties } ]
+// but are now { message: 'messaeg text', ... properties }.
+// This makes sure legacy events in the reducer are the expected data types.
+const fixLegacyEvent = event => {
```

Comment:
I was going to delete before merging. It's probably safe to do now, just requires clearing cached state.